diff --git a/src/uu/nohup/src/platform/unix.rs b/src/uu/nohup/src/platform/unix.rs
--- a/src/uu/nohup/src/platform/unix.rs
+++ b/src/uu/nohup/src/platform/unix.rs
@@ -17,7 +17,7 @@ use crate::find_stdout;
 
 #[derive(Debug, ThisError)]
 enum PlatformError {
-    #[cfg(target_vendor = "apple")]
+    #[cfg(all(target_vendor = "apple", not(target_os = "ios")))]
     #[error("{}", translate!("nohup-error-cannot-detach"))]
     CannotDetach,
 
@@ -38,7 +38,7 @@ pub(crate) fn prepare() -> UResult<()> {
 
     unsafe { libc::signal(libc::SIGHUP, libc::SIG_IGN) };
 
-    #[cfg(target_vendor = "apple")]
+    #[cfg(all(target_vendor = "apple", not(target_os = "ios")))]
     if unsafe { !_vprocmgr_detach_from_console(0).is_null() } {
         return Err(PlatformError::CannotDetach.into());
     }
@@ -86,7 +86,7 @@ fn replace_fds() -> UResult<()> {
     Ok(())
 }
 
-#[cfg(target_vendor = "apple")]
+#[cfg(all(target_vendor = "apple", not(target_os = "ios")))]
 unsafe extern "C" {
     fn _vprocmgr_detach_from_console(flags: u32) -> *const core::ffi::c_int;
 }
